src/main/java/com/example/schooloperationsystem/service/impl/HeadMasterServiceImpl.java

```diff
@@ -9,28 +9,39 @@
 import com.example.schooloperationsystem.service.HeadMasterService;
 import com.example.schooloperationsystem.service.params.CreateHeadMasterParams;
 import jakarta.transaction.Transactional;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Service;
 import com.example.schooloperationsystem.entity.Teacher;
 
 import java.util.List;
 
+@Slf4j
 @Service
+@AllArgsConstructor
 class HeadMasterServiceImpl implements HeadMasterService {
+
     private final HeadMasterRepository repository;
 
-    public HeadMasterServiceImpl(HeadMasterRepository repository) {
-        this.repository = repository;
-    }
+    private static final Logger logger = LoggerFactory.getLogger(HeadMasterServiceImpl.class);
```

Comment:
becasue we already have @Slf4j lombok annotation, there is no need to define logger here